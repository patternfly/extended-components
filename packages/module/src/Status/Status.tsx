import { css } from '@emotion/react';
import * as React from 'react';
import { Button, ButtonVariant, Flex, FlexItem, Icon, Popover, PopoverPosition, PopoverProps, Content, ContentVariants, } from '@patternfly/react-core';

export const StatusVariant = {
  link: 'link',
  popover: 'popover',
  plain: 'plain'
} as const;

export type StatusVariant = typeof StatusVariant[keyof typeof StatusVariant];

export const IconStatus = {
  custom: 'custom',
  info: 'info',
  success: 'success',
  warning: 'warning',
  danger: 'danger'
} as const;

export type IconStatus = typeof IconStatus[keyof typeof IconStatus];

export interface StatusProps extends React.PropsWithChildren {
  /** Status label text */
  label?: string;
  /** Description to be displayed under the label */
  description?: string;
  /** If true, only displays icon */
  iconOnly?: boolean;
  /** Variant of the status component to be displayed */
  variant?: StatusVariant;
  /** Status icon */
  icon?: React.ReactElement;
  /** Status to control icon color */
  status: IconStatus;
  /** Icon title for accessibility */
  iconTitle?: string;
  /** Custom OUIA ID */
  ouiaId?: string | number;
  /** Props for the optional popover */
  popoverProps?: PopoverProps;
  /** Optional link variant onClick callback */
  onClick?: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
}

const styles = {
  icon: css`
    margin: 0;
    align-self: flex-start;
  `,
  statusLabel: css`
    line-height: var(--pf-t--global--spacer--md);
  `,
  statusDescription: css`
    color: var(--pf-t--color--gray--50);
  `
};

export const Status: React.FC<StatusProps> = ({ variant = StatusVariant.plain, label, children, iconOnly, icon, status, iconTitle, ouiaId = 'Status', popoverProps, onClick, description, ...props }: StatusProps) => {

  if (iconOnly && !iconTitle) {
    // eslint-disable-next-line no-console
    console.warn('iconOnly is true but no iconTitle is provided. Please provide a descriptive iconTitle for accessibility.');
  }

  const statusBody = (
    <Flex title={label} alignItems={{ default: 'alignItemsCenter' }} {...props}>
      {icon && (
        <FlexItem css={styles.icon}>
          <Icon className='pf-v6-u-mr-sm' status={status} title={iconTitle ?? status} data-ouia-component-id={`${ouiaId}-icon`}>
            {icon}
          </Icon>
        </FlexItem>
      )}
      {!iconOnly && (
        <FlexItem>
          <Content ouiaId={`${ouiaId}-label`} 
            css={styles.statusLabel} style={{ marginBlockEnd: 0 }}>{label}</Content>
          {description && <Content component={ContentVariants.small} ouiaId={`${ouiaId}-description`} css={styles.statusDescription}>{description}</Content>}
        </FlexItem>
      )}
    </Flex>
  );

  if (variant === StatusVariant.link) {
    return (
      <Button variant={ButtonVariant.link} title={label} onClick={onClick} ouiaId={`${ouiaId}-link-icon`}>
        {statusBody}
      </Button>
    );
  }

  if (variant === StatusVariant.popover) {
    return (
      <Popover
        position={PopoverPosition.right}
        headerContent={label}
        bodyContent={children}
        aria-label={label}
        data-ouia-component-id={`${ouiaId}-popover`}
        {...popoverProps}
      >
        <Button variant={ButtonVariant.link} isInline style={{ textDecoration: 'none' }} ouiaId={`${ouiaId}-popover-icon`}>
          {statusBody}
        </Button>
      </Popover>
    );
  }

  return statusBody;
};

export default Status;
