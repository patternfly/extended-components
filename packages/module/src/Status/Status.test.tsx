import React from 'react';
import { render } from '@testing-library/react';
import Status, { IconStatus, StatusVariant } from './Status';
import { BanIcon, CheckCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons';
import { Button, ButtonSize, ButtonVariant } from '@patternfly/react-core';

describe('Status component', () => {
  it('should render correctly', () => {
    expect(render(<Status status={IconStatus.warning} label='Warning' icon={<ExclamationTriangleIcon/>}/>)).toMatchSnapshot();
  });

  it('should render iconOnly correctly', () => {
    expect(render(<Status iconOnly status={IconStatus.warning} iconTitle='1 security issue found' label='Warning' icon={<ExclamationTriangleIcon/>} />)).toMatchSnapshot();
  });

  it('should render correctly with description', () => {
    expect(render(<Status status={IconStatus.warning} label='Warning' description='1 issue found' icon={<ExclamationTriangleIcon/>}/>)).toMatchSnapshot();
  });

  it('should render correctly link', () => {
    // eslint-disable-next-line no-console
    expect(render(<Status status={IconStatus.success} variant={StatusVariant.link} label='Ready' onClick={() => console.log('Link status clicked')} icon={<CheckCircleIcon/>}/>)).toMatchSnapshot();
  });

  it('should render correctly popover', () => {
    expect(render(<Status
      status={IconStatus.danger}
      variant={StatusVariant.popover} 
      label='Not Ready' 
      icon={<BanIcon/>}
      popoverProps={{ 
        bodyContent: 'Example state description', 
        footerContent: <Button size={ButtonSize.sm} variant={ButtonVariant.link} isInline>Action</Button> 
      }}
    />)).toMatchSnapshot();
  });
});
