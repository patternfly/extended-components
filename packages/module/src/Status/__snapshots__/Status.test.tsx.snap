// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-v6-l-flex pf-m-align-items-center"
        title="Warning"
      >
        <div
          class="icon-0-2-1"
        >
          <span
            class="pf-v6-c-icon pf-v6-u-mr-md"
            data-ouia-component-id="Status-icon"
            title="warning"
          >
            <span
              class="pf-v6-c-icon__content pf-m-warning"
            >
              <svg
                aria-hidden="true"
                class="pf-v6-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          class=""
        >
          <p
            class="pf-v6-c-content--p statusLabel-0-2-2"
            data-ouia-component-id="Status-label"
            data-ouia-component-type="PF6/Text"
            data-ouia-safe="true"
            data-pf-content="true"
            style="margin-block-end: 0;"
          >
            Warning
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-v6-l-flex pf-m-align-items-center"
      title="Warning"
    >
      <div
        class="icon-0-2-1"
      >
        <span
          class="pf-v6-c-icon pf-v6-u-mr-md"
          data-ouia-component-id="Status-icon"
          title="warning"
        >
          <span
            class="pf-v6-c-icon__content pf-m-warning"
          >
            <svg
              aria-hidden="true"
              class="pf-v6-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 576 512"
              width="1em"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        class=""
      >
        <p
          class="pf-v6-c-content--p statusLabel-0-2-2"
          data-ouia-component-id="Status-label"
          data-ouia-component-type="PF6/Text"
          data-ouia-safe="true"
          data-pf-content="true"
          style="margin-block-end: 0;"
        >
          Warning
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status component should render correctly link 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-disabled="false"
        class="pf-v6-c-button pf-m-link"
        data-ouia-component-id="Status-link-icon"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        title="Ready"
        type="button"
      >
        <div
          class="pf-v6-l-flex pf-m-align-items-center"
          title="Ready"
        >
          <div
            class="icon-0-2-1"
          >
            <span
              class="pf-v6-c-icon pf-v6-u-mr-md"
              data-ouia-component-id="Status-icon"
              title="success"
            >
              <span
                class="pf-v6-c-icon__content pf-m-success"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class=""
          >
            <p
              class="pf-v6-c-content--p statusLabel-0-2-2"
              data-ouia-component-id="Status-label"
              data-ouia-component-type="PF6/Text"
              data-ouia-safe="true"
              data-pf-content="true"
              style="margin-block-end: 0;"
            >
              Ready
            </p>
          </div>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-disabled="false"
      class="pf-v6-c-button pf-m-link"
      data-ouia-component-id="Status-link-icon"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      title="Ready"
      type="button"
    >
      <div
        class="pf-v6-l-flex pf-m-align-items-center"
        title="Ready"
      >
        <div
          class="icon-0-2-1"
        >
          <span
            class="pf-v6-c-icon pf-v6-u-mr-md"
            data-ouia-component-id="Status-icon"
            title="success"
          >
            <span
              class="pf-v6-c-icon__content pf-m-success"
            >
              <svg
                aria-hidden="true"
                class="pf-v6-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          class=""
        >
          <p
            class="pf-v6-c-content--p statusLabel-0-2-2"
            data-ouia-component-id="Status-label"
            data-ouia-component-type="PF6/Text"
            data-ouia-safe="true"
            data-pf-content="true"
            style="margin-block-end: 0;"
          >
            Ready
          </p>
        </div>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status component should render correctly popover 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: contents;"
      >
        <button
          aria-disabled="false"
          class="pf-v6-c-button pf-m-link pf-m-inline"
          data-ouia-component-id="Status-popover-icon"
          data-ouia-component-type="PF6/Button"
          data-ouia-safe="true"
          style="text-decoration: none;"
          type="button"
        >
          <div
            class="pf-v6-l-flex pf-m-align-items-center"
            title="Not Ready"
          >
            <div
              class="icon-0-2-1"
            >
              <span
                class="pf-v6-c-icon pf-v6-u-mr-md"
                data-ouia-component-id="Status-icon"
                title="danger"
              >
                <span
                  class="pf-v6-c-icon__content pf-m-danger"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class=""
            >
              <p
                class="pf-v6-c-content--p statusLabel-0-2-2"
                data-ouia-component-id="Status-label"
                data-ouia-component-type="PF6/Text"
                data-ouia-safe="true"
                data-pf-content="true"
                style="margin-block-end: 0;"
              >
                Not Ready
              </p>
            </div>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: contents;"
    >
      <button
        aria-disabled="false"
        class="pf-v6-c-button pf-m-link pf-m-inline"
        data-ouia-component-id="Status-popover-icon"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        style="text-decoration: none;"
        type="button"
      >
        <div
          class="pf-v6-l-flex pf-m-align-items-center"
          title="Not Ready"
        >
          <div
            class="icon-0-2-1"
          >
            <span
              class="pf-v6-c-icon pf-v6-u-mr-md"
              data-ouia-component-id="Status-icon"
              title="danger"
            >
              <span
                class="pf-v6-c-icon__content pf-m-danger"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class=""
          >
            <p
              class="pf-v6-c-content--p statusLabel-0-2-2"
              data-ouia-component-id="Status-label"
              data-ouia-component-type="PF6/Text"
              data-ouia-safe="true"
              data-pf-content="true"
              style="margin-block-end: 0;"
            >
              Not Ready
            </p>
          </div>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status component should render correctly with description 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-v6-l-flex pf-m-align-items-center"
        title="Warning"
      >
        <div
          class="icon-0-2-1"
        >
          <span
            class="pf-v6-c-icon pf-v6-u-mr-md"
            data-ouia-component-id="Status-icon"
            title="warning"
          >
            <span
              class="pf-v6-c-icon__content pf-m-warning"
            >
              <svg
                aria-hidden="true"
                class="pf-v6-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          class=""
        >
          <p
            class="pf-v6-c-content--p statusLabel-0-2-2"
            data-ouia-component-id="Status-label"
            data-ouia-component-type="PF6/Text"
            data-ouia-safe="true"
            data-pf-content="true"
            style="margin-block-end: 0;"
          >
            Warning
          </p>
          <small
            class="pf-v6-c-content--small statusDescription-0-2-3"
            data-ouia-component-id="Status-description"
            data-ouia-component-type="PF6/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            1 issue found
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-v6-l-flex pf-m-align-items-center"
      title="Warning"
    >
      <div
        class="icon-0-2-1"
      >
        <span
          class="pf-v6-c-icon pf-v6-u-mr-md"
          data-ouia-component-id="Status-icon"
          title="warning"
        >
          <span
            class="pf-v6-c-icon__content pf-m-warning"
          >
            <svg
              aria-hidden="true"
              class="pf-v6-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 576 512"
              width="1em"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        class=""
      >
        <p
          class="pf-v6-c-content--p statusLabel-0-2-2"
          data-ouia-component-id="Status-label"
          data-ouia-component-type="PF6/Text"
          data-ouia-safe="true"
          data-pf-content="true"
          style="margin-block-end: 0;"
        >
          Warning
        </p>
        <small
          class="pf-v6-c-content--small statusDescription-0-2-3"
          data-ouia-component-id="Status-description"
          data-ouia-component-type="PF6/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          1 issue found
        </small>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status component should render iconOnly correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-v6-l-flex pf-m-align-items-center"
        title="Warning"
      >
        <div
          class="icon-0-2-1"
        >
          <span
            class="pf-v6-c-icon pf-v6-u-mr-md"
            data-ouia-component-id="Status-icon"
            title="1 security issue found"
          >
            <span
              class="pf-v6-c-icon__content pf-m-warning"
            >
              <svg
                aria-hidden="true"
                class="pf-v6-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 576 512"
                width="1em"
              >
                <path
                  d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-v6-l-flex pf-m-align-items-center"
      title="Warning"
    >
      <div
        class="icon-0-2-1"
      >
        <span
          class="pf-v6-c-icon pf-v6-u-mr-md"
          data-ouia-component-id="Status-icon"
          title="1 security issue found"
        >
          <span
            class="pf-v6-c-icon__content pf-m-warning"
          >
            <svg
              aria-hidden="true"
              class="pf-v6-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 576 512"
              width="1em"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
